### DISCUSSION TITLE
Exporting methods for the collection within variables.

### DISCUSSION THREAD

**Discussion by hayawata3626** [lukasoppermann/design-tokens#270] [Q&A] [answered]
Within Figma's variables, there exists a feature called "collection". What are your thoughts on allowing collections to be exportable?

**Use Case:**
While engineers are in the development phase, designers often update design tokens. There is a potential risk when designs, both finalized and in draft status, get exported together. Engineers might mistakenly refer to the draft designs.

**collectionA** => for draft designs
**collectionB** => for finalized designs

Ideally, when exporting, it would be beneficial if one could select 'collectionB' using a checkbox or similar functionality. Another approach could be to name the collections with terms like "draft", ensuring that the code does not refer to properties named "draft".

If there are any other effective methods or recommendations, it would be appreciated if shared.

**Comment by lukasoppermann** [comment, marked as answer]
Hey, the easiest way would be to use a `_` or `.` prefix. This is a Figma collection.

A good feature would be to not export collections that start with one of the two prefixes.

**Reply by hayawata3626** [reply]
Thank you for your response!
I tried, but the result turned out as follows.

**Setting**
<img width="285" alt="スクリーンショット 2023-09-13 14 11 13" src="https://github.com/lukasoppermann/design-tokens/assets/15213369/a6302ab2-89bc-42c2-977f-602f85814363">

**Collection**
<img width="672" alt="スクリーンショット 2023-09-13 14 12 10" src="https://github.com/lukasoppermann/design-tokens/assets/15213369/78436f0a-f1ef-4424-87c3-7d57fb4e0f3f">


**JSON**
<img width="413" alt="スクリーンショット 2023-09-13 14 12 42" src="https://github.com/lukasoppermann/design-tokens/assets/15213369/7604bab4-0fbb-476a-be7d-6e95befc0afe">

The collection named "_test" is currently included.
Do we need to add this feature?

**Reply by lukasoppermann** [reply]
Yes, sorry this is what I meant. We need to add this feature.

**Reply by hayawata3626** [reply]
So, this setting doesn't apply to the collection in the "variables". Understood.

<img width="285" alt="スクリーンショット 2023-09-13 14 11 13" src="https://github.com/lukasoppermann/design-tokens/assets/15213369/2cf99109-594f-4425-aa46-7a94e722eec6">



I'll try to add the feature. 
~~Should we allow users to specify prefixes they want to ignore on the settings page? Or should we not have any settings on the interface and automatically exclude prefixes with "_" or "." from the export target?~~

- Add an explanation to the UI that this also applies to the collection.
- Modify the logic (do not export if the collection contains "." or "_").

Is this what you have in mind?

**Reply by lukasoppermann** [reply]
Maybe we reuse the exclusion prefix setting for that?

**Reply by hayawata3626** [reply]
At the first, I try to reuse the exclusion prefix setting.
If I cannot reuse the exclusion prefix setting, I will add the ui to set  exclusion prefix for the collection.

**Reply by hayawata3626** [reply]
I added the [PR](https://github.com/lukasoppermann/design-tokens/pull/275)

**Reply by hayawata3626** [reply]
Thank you marge, I will resolve the discussion!
